### PR TITLE
solution

### PR DESCRIPTION
**Pull request by olenitut**
[DEMO LINK](https://olenitut.github.io/react_movies-list-fetch-movies/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -3,9 +3,55 @@ import './App.scss';
...
+      const newMovie: Movie = {
+        title: data.Title,
+        description: data.Plot,
+        imgUrl: data.Poster === 'N/A' ? defaultPicture : data.Poster,
+        imdbId: data.imdbID,
+        imdbUrl: `https://www.imdb.com/title/${data.imdbID}`,
+      };
+
+      setCurrentMovie(newMovie);
+    }).finally(() => setIsLoading(false));
```
Comment (OlEzhA0) [inline src/App.tsx:35]: try to use `async/await` syntax, it’s more preferable

File: src/App.tsx [hunk 2]
```diff
@@ -3,9 +3,55 @@ import './App.scss';
...
+
+    getMovie(userQuery).then(data => {
+      if ('Error' in data) {
+        setCurrentMovie(null);
+
+        return;
+      }
+
+      // eslint-disable-next-line max-len
+      const defaultPicture = 'https://via.placeholder.com/360x270.png?text=no%20preview';
```
Comment (OlEzhA0) [inline src/App.tsx:24]: define this constant outside your component

File: src/App.tsx [hunk 3]
```diff
@@ -3,9 +3,55 @@ import './App.scss';
...
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [currentMovie, setCurrentMovie] = useState<null | Movie>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const findMovieHandler = (userQuery: string) => {
+    setIsLoading(true);
+
+    getMovie(userQuery).then(data => {
```
Comment (OlEzhA0) [inline src/App.tsx:16]: where do you `catching` your error?

File: src/App.tsx [hunk 4]
```diff
@@ -3,9 +3,55 @@ import './App.scss';
...
+  const addMovieHandler = () => {
+    setMovies(prevMovies => {
+      const newMovies = [...prevMovies];
+
+      if (currentMovie) {
+        if (newMovies.find(mov => mov.imdbId === currentMovie.imdbId)) {
+          return newMovies;
+        }
+
+        newMovies.push(currentMovie);
```
Comment (OlEzhA0) [inline src/App.tsx:47]:
```suggestion
        return [...prevMovies, currentMovie]
```

File: src/components/FindMovie/FindMovie.tsx [hunk 5]
```diff
@@ -1,10 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  onFind: (arg0: string) => void;
```
Comment (OlEzhA0) [inline src/components/FindMovie/FindMovie.tsx:8]:
```suggestion
  onFind: (value: string) => void;
```

File: src/components/FindMovie/FindMovie.tsx [hunk 6]
```diff
@@ -1,10 +1,45 @@
...
+  movie: Movie | null;
+  onAdd: () => void;
+  isLoading: boolean
+};
+
+export const FindMovie: React.FC<Props> = ({
```
Comment (OlEzhA0) [inline src/components/FindMovie/FindMovie.tsx:14]: nb: destruct FC from react

**Review by OlEzhA0 — CHANGES_REQUESTED** [review]
Hey! Please check my comments
ping me in fe_chat if you have any questions

File: src/App.tsx (hunk 3 above)
Comment (olenitut) [reply src/App.tsx:16]: here     if ('Error' in data) {
      setCurrentMovie(null);
      setIsLoading(false);

      return;
    }

Comment (olenitut) [reply src/App.tsx:16]: check what the api function retruns for errors

File: src/components/FindMovie/FindMovie.tsx (hunk 6 above)
Comment (olenitut) [reply src/components/FindMovie/FindMovie.tsx:14]: no, i prefere this variant

Comment (witflash) [reply src/components/FindMovie/FindMovie.tsx:14]: Why do you prefer this variant? :)
It's ok, but what is the reason? )

File: src/components/FindMovie/FindMovie.tsx [hunk 7]
```diff
@@ -1,10 +1,45 @@
...
+  onFind, movie, onAdd, isLoading,
+}) => {
+  const [enteredValue, setEnteredValue] = useState('');
+  const [attempteSearch, setAttemptedSearch] = useState(false);
```
Comment (witflash) [inline src/components/FindMovie/FindMovie.tsx:18]: You have a typo in `attemptedSearch` I think

File: src/App.tsx (hunk 4 above)
Comment (witflash) [reply src/App.tsx:49]: @olenitut What about this comment?

**Review by witflash — CHANGES_REQUESTED** [review]
Well done!
But don't leave unresolved comments, please.

File: src/components/FindMovie/FindMovie.tsx (hunk 6 above)
Comment (olenitut) [reply src/components/FindMovie/FindMovie.tsx:14]: I believe that when i use syntax React.FC, it is much more readable and easy to understand where this variable comes from. The same goes with destructuring props. I get a lot of comments from mentors that i need to destructure them but i believe that when i use props.children, the code becomes much more readable for me and others

Comment (witflash) [reply src/components/FindMovie/FindMovie.tsx:14]: You may change your mind when you will work on a huge project :)
But ok )

P. S. Destructuring is a trending best practice. So you should remember this.

**Review by witflash — APPROVED** [review]

**Pull request closed without merging**